mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -272,8 +295,12 @@ impl ZRpcOps for NativeClient {
 }
 
 /// A wrapper for the SQLite connection to the block cache database.
+#[cfg(not(target_arch = "wasm32"))]
 pub struct BlockDb(Connection);
+#[cfg(target_arch = "wasm32")]
+pub struct BlockDb(String);
 
+#[cfg(not(target_arch = "wasm32"))]
```

Comment:
please follow BlockHeaderStorageOps example https://github.com/KomodoPlatform/atomicDEX-API/blob/d01dd41e91ccf7e353a4cdb4e2135bdeb8dfc3de/mm2src/mm2_bitcoin/spv_validation/src/storage.rs#L87
by creating a common trait that should be implemented for both `struct BlockDb(Connection)` and `pub struct BlockDb(String);`. Shouldn't `pub struct BlockDb(String);` use `IndexedDb` and not a `String` ? You can change these structs as you see fit.

Reply:
`BlockDb` can't be implemented like the way `BlockHeaderStorageOps` is done because `BlockDb` additionally needs to implement a trait with a generic function.
https://github.com/KomodoPlatform/atomicDEX-API/blob/46535e882896e10aa16f3d6364007ba2a8c0cc3d/mm2src/coins/z_coin/z_rpc.rs#L363-L372

which makes doing these impossible with the current rust

https://github.com/KomodoPlatform/atomicDEX-API/blob/46535e882896e10aa16f3d6364007ba2a8c0cc3d/mm2src/coins/utxo/utxo_block_header_storage/mod.rs#L19-L21

Meanwhile, I already complete the implementation structure in another branch
https://github.com/KomodoPlatform/atomicDEX-API/blob/daa1b650b0ae319c79923ee0f651f889a7a438cc/mm2src/coins/z_coin/storage/blockdb/mod.rs#L30-L38

Reply:
> Meanwhile, I already completed the implementation structure in another branch, so I got it covered

May be it worth adding this implementation in this PR with todos if you have it? So we can see the whole picture.

Reply:
Ok cool, after Omar's comment

Reply:
> BlockDb can't be implemented like the way BlockHeaderStorageOps is done because BlockDb additionally needs to implement a trait with a generic function.

BlockDb can implement multiple traits, I don't see the problem. It doesn't have to be exactly like `BlockHeaderStorageOps`/`BlockHeaderStorage` but I meant to take it as an example. Another example is the `LightningDB` trait which I plan to implement for IndexedDB later and have a common interface. https://github.com/KomodoPlatform/atomicDEX-API/blob/a1fc8f7c92819f68c0dcdf7b1c650d86cdf945ad/mm2src/coins/lightning/ln_db.rs#L200

> May be it worth adding this implementation in this PR with todos if you have it? So we can see the whole picture.

As discussed with @borngraced on DM, he will add them to this PR, as for the above suggestion about following `BlockHeaderStorageOps`/`LightningDB`/etc.. you can try that in the next PRs since it will be tricky dealing with generics.